org.eclipse.lsp4e/src/org/eclipse/lsp4e/server/ProcessInfo.java

```diff
@@ -0,0 +1,23 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Vegard IT GmbH and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Sebastian Thomschke (Vegard IT GmbH) - initial implementation
+ *******************************************************************************/
+package org.eclipse.lsp4e.server;
+
+import org.eclipse.jdt.annotation.Nullable;
+
+public interface ProcessInfo {
```

Comment:
I don't think we need such an interface, using a adapter to ProcessHandle would reduce coupling and code to maintain while augmenting the capability of the feature.